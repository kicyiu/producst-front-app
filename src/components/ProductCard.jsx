import React from 'react';

function ProductCard(props) {
    return (
        <div className="card puntero">
            <img src={`http://${props.image}`} className="card-img-top" alt="..."></img>
            <div className="card-body">
                <p> 
                    <b className="card-title">{props.brand}&nbsp;</b>
                    <span className="card-desc">
                        {props.description}
                    </span>
                </p>
                {props.deal ? (
                    <div>
                        <p className="card-text">
                            ${props.price}&nbsp;
                            <span className="badge badge-pill badge-primary custom-badge">
                                50%
                            </span>
                        </p>
                        <p className="current-price">
                            ${props.price * 2}
                        </p>
                    </div>
                ): (
                    <p className="card-text">
                        ${props.price}
                    </p>
                )}
            </div>
        </div>
    );
}

export default ProductCard;