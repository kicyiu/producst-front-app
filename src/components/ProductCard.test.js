import React from 'react';
import renderer from 'react-test-renderer';

import ProductCard from './ProductCard';

it('renders ProductCard correctly when there is a palindrono search', () => {
    const tree = renderer.create(<ProductCard deal={true}/>).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders ProductCard correctly when there is not palindrono search', () => {
    const tree = renderer.create(<ProductCard deal={false}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });