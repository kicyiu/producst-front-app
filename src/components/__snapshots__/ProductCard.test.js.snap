// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ProductCard correctly when there is a palindrono search 1`] = `
<div
  className="card puntero"
>
  <img
    alt="..."
    className="card-img-top"
    src="http://undefined"
  />
  <div
    className="card-body"
  >
    <p>
      <b
        className="card-title"
      >
         
      </b>
      <span
        className="card-desc"
      />
    </p>
    <div>
      <p
        className="card-text"
      >
        $
         
        <span
          className="badge badge-pill badge-primary custom-badge"
        >
          50%
        </span>
      </p>
      <p
        className="current-price"
      >
        $
        NaN
      </p>
    </div>
  </div>
</div>
`;

exports[`renders ProductCard correctly when there is not palindrono search 1`] = `
<div
  className="card puntero"
>
  <img
    alt="..."
    className="card-img-top"
    src="http://undefined"
  />
  <div
    className="card-body"
  >
    <p>
      <b
        className="card-title"
      >
         
      </b>
      <span
        className="card-desc"
      />
    </p>
    <p
      className="card-text"
    >
      $
    </p>
  </div>
</div>
`;
